backbone.js

```diff
@@ -1088,8 +1088,8 @@
     },
 
     // Define how to uniquely identify models in the collection.
-    modelId: function(attrs) {
-      return attrs[this.model.prototype.idAttribute || 'id'];
+    modelId: function(attrs, model) {
+      return attrs[(model && model.idAttribute) || this.model.prototype.idAttribute || 'id'];
```

Comment:
Although, that conflicting id concern is assuming polymorphic models can have conflicting ids, an assumption that Backbone already seems to ignore: http://backbonejs.org/#Collection-model